src/main/java/uk/ac/sanger/sccp/stan/service/releasefile/ReleaseFileService.java

```diff
@@ -414,6 +427,181 @@ public void loadSectionDate(Collection<ReleaseEntry> entries, Ancestry ancestry)
+            final Integer lwId = entry.getLabware().getId();
+            Operation op = lwOp.get(lwId);
+            if (op==null) {
+                continue;
+            }
+            entry.setXeniumStart(op.getPerformed());
+            List<Roi> rois = opIdRoi.get(op.getId());
+            if (!nullOrEmpty(rois) && entry.getSlot()!=null && entry.getSample()!=null) {
+                rois.stream()
+                        .filter(roi -> roi.getSampleId().equals(entry.getSample().getId()) && roi.getSlotId().equals(entry.getSlot().getId()))
+                        .findAny()
+                        .ifPresent(roi -> entry.setXeniumRoi(roi.getRoi()));
+            }
+            List<LabwareNote> notes = opIdNotes.get(op.getId());
+            if (nullOrEmpty(notes)) {
+                continue;
+            }
+            for (LabwareNote note : notes) {
+                if (!lwId.equals(note.getLabwareId())) {
+                    continue;
+                }
+                if (note.getName().equalsIgnoreCase(AnalyserServiceImp.LOT_NAME)) {
+                    entry.setXeniumReagentLot(note.getValue());
+                } else if (note.getName().equalsIgnoreCase(AnalyserServiceImp.POSITION_NAME)) {
+                    entry.setXeniumCassettePosition(note.getValue());
+                } else if (note.getName().equalsIgnoreCase(AnalyserServiceImp.RUN_NAME)) {
+                    entry.setXeniumRun(note.getValue());
+                }
+            }
+        }
+    }
+
+    /**
+     * Loads timestamps and comments from Xenium QC ops.
+     */
+    public void loadXeniumQC(Collection<ReleaseEntry> entries, Set<Integer> slotIds) {
+        OperationType opType = opTypeRepo.getByName("Xenium QC");
+        List<Operation> ops = opRepo.findAllByOperationTypeAndDestinationSlotIdIn(opType, slotIds);
+        if (ops.isEmpty()) {
+            return;
+        }
+        List<Integer> opIds = ops.stream().map(Operation::getId).collect(toList());
+        Map<Integer, List<OperationComment>> opIdComs = opComRepo.findAllByOperationIdIn(opIds).stream()
+                .collect(groupingBy(OperationComment::getOperationId));
+        Map<Integer, Operation> lwIdOps = labwareIdToOp(ops);
+        for (ReleaseEntry entry : entries) {
+            Integer lwId = entry.getLabware().getId();
+            Operation op = lwIdOps.get(lwId);
+            if (op==null) {
+                return;
+            }
+            entry.setXeniumEnd(op.getPerformed());
+            List<OperationComment> opcoms = opIdComs.get(op.getId());
+            if (nullOrEmpty(opcoms)) {
+                return;
+            }
+            String commentText = opcoms.stream()
+                    .filter(oc -> lwId.equals(oc.getLabwareId()))
+                    .map(oc -> oc.getComment().getText())
+                    .map(s -> s.endsWith(".") ? s : (s + "."))
+                    .collect(joining(" "));
+            entry.setXeniumComment(commentText);
+        }
```

Comment:
I noticed that the loadProbeHybridisationQC and loadXeniumQC methods share a similar piece of code for joining comments. It might be a good idea to extract this common code into a separate method to avoid duplication.

Reply:
OK. I've put the comment-joining logic into a method.